File: frontend/projects/api/src/lib/lifecycle-progress.ts
export interface LifecycleProgress {
  activeStage: string;
  completedStages: string[];
  createdAt: string;
  processingStartedAt: string | null;
  validationCompletedAt: string | null;
  reviewedAt: string | null;
  publishedAt: string | null;
  retiredAt: string | null;
  isStale: boolean;
  isActionRequired: boolean;
  failureCode: string | null;
  failureDetail: string | null;
}

File: backend/src/RepoFluent.Application/LifecycleProgress.cs
namespace RepoFluent.Application;

public sealed record LifecycleProgress(
    string ActiveStage,
    IReadOnlyList<string> CompletedStages,
    DateTimeOffset CreatedAt,
    DateTimeOffset? ProcessingStartedAt,
    DateTimeOffset? ValidationCompletedAt,
    DateTimeOffset? ReviewedAt,
    DateTimeOffset? PublishedAt,
    DateTimeOffset? RetiredAt,
    bool IsStale,
    bool IsActionRequired,
    string? FailureCode,
    string? FailureDetail);
